src/main/java/uk/gov/hmcts/reform/notificationservice/service/NotificationMessageParser.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.hmcts.reform.notificationservice.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.azure.servicebus.MessageBody;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+import uk.gov.hmcts.reform.notificationservice.exception.InvalidMessageException;
+import uk.gov.hmcts.reform.notificationservice.model.request.incomming.NotificationMsg;
+
+import java.io.IOException;
+import java.util.List;
+
+@Service
+public class NotificationMessageParser {
+
+    private static final Logger logger = LoggerFactory.getLogger(NotificationMessageParser.class);
+
+    private final ObjectMapper objectMapper;
+
+    public NotificationMessageParser(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    public NotificationMsg parse(MessageBody messageBody) {
+        try {
+            NotificationMsg notificationMsg =
+                objectMapper.readValue(getBinaryData(messageBody), NotificationMsg.class);
+            logger.info(
+                "Parsed notification message, Zip File Name: {}, Error Code: {}, "
+                    + "Jurisdiction: {}, PO Box: {}, Service: {}, Document Control Number: {}",
+                notificationMsg.zipFileName,
+                notificationMsg.errorCode,
+                notificationMsg.jurisdiction,
+                notificationMsg.poBox,
+                notificationMsg.service,
+                notificationMsg.documentControlNumber
+            );
+
+            return notificationMsg;
+        } catch (IOException exc) {
+            throw new InvalidMessageException(exc);
```

Comment:
Suggestion: while this gives us the information that we need, adding a message (in the exception being thrown) that we failed to parse the message body could be helpful.